pkg/apis/core/types.go

```diff
@@ -5176,8 +5176,8 @@ const (
 // TopologySpreadConstraint specifies how to spread matching pods among the given topology.
 type TopologySpreadConstraint struct {
 	// MaxSkew describes the degree to which pods may be unevenly distributed.
-	// It's the maximum permitted difference between the number of matching pods in
-	// any two topology domains of a given topology type.
+	// When `whenUnsatisfiable=DoNotSchedule`, it is the maximum permitted difference
+	// between the number of matching pods in any two topology domains of a given topology type.
```

Comment:
I just realized the wording "... in any two topology domains ..." is ambiguous. As internally the logic is checking the difference between **targeting** skew and global skew.
For instance, suppose we have 3 zones with matching Pods as `3, 1, 0` and maxSkew=1. Following current comments, it seems we couldn't schedule another Pod as current global skew is 3, even we put it to zone3, it can only become 2 (which violates given maxSkew=1). However, it's not the real semantics. The real one is: we can schedule the Pod to zone3, as we're checking the actual skew (which is 0) on target topology (zone3). And the Pod cannot be scheduled to zone1/zone2 - the corresponding skew diff would be 4/2.

So I'd suggest to update it to:

```
// between the number of matching pods in the target topology and the one with minimum
// matching number, per  given topology type.
```

or more clear wordings.

Reply:
Done